### PR TITLE
Add support for custom annotations to metadata annotations and bundle.Dockerfile

### PR DESCRIPTION
**Pull request by redhatrises**
**Description of the change:**

- Add capability for a project to add custom labels to bundle.Dockerfile and metadata annotations.yaml.
  This can be done by creating `config/metadata/annotations.yaml` which will allow changes to happen
  to annotations.yaml and bundle.Dockerfile in the operator-sdk code independently of a project's desired
  custom annotations.

**Motivation for the change:**

`operator-sdk generate bundle -q --overwrite --version $(VERSION) $(BUNDLE_METADATA_OPTS)` consistently overwrites any custom annotations applied to `bundle/metadata/annotations.yaml` and `bundle.Dockerfile`. To add any custom annotations, you have to remove `--overwrite` and manually maintain these files which also means that `make bundle` will no longer make any changes that a project may want to happen to `bundle/metadata/annotations.yaml` and `bundle.Dockerfile`. This PR allows the best of both worlds where a project can inject custom labels simply by creating 
`config/metadata/annotations.yaml`, and the operator-sdk can continue if necessary to update/change the base settings as needed.

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [ ] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [ ] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

**Review by camilamacedo86 — COMMENTED** [review]
Could you please: Add a new changelog fragment in changelog/fragments (see [changelog/fragments/00-template.yaml](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))

Also, what is the goal of this PR? Is ensure that we will sync all annotations in the file to the bundle.Dockerfile? If yes, why do we need a flag for that? Why not sync all of them? See: https://github.com/operator-framework/operator-sdk/issues/5002

**Comment by redhatrises** [comment]
> Could you please: Add a new changelog fragment in changelog/fragments (see [changelog/fragments/00-template.yaml](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
> 
> Also, what is the goal of this PR? Is ensure that we will sync all annotations in the file to the bundle.Dockerfile? If yes, why do we need a flag for that? Why not sync all of them? See: #5002

What @isimluk replied with is correct, but this isn't about "all" annotations as this is about custom annotations which in this case is requirements around Red Hat Marketplace.

**Comment by camilamacedo86** [comment]
Hi @isimluk, @redhatrises,

>  What @isimluk replied with is correct, but this isn't about "all" annotations as this is about custom annotations which in this case is requirements around Red Hat Marketplace.

- why not ensure that all annotations in bundle/annotations.yml file end up as a LABEL in the bundle.Dockerfile? 
- why we would need to filter or limit what goes or not? 

It was discussed in the triage meeting with https://github.com/operator-framework/operator-sdk/issues/5002 and @estroz (who implement the feature at the time) just shared that he thought that we should sync only the doc ones in https://github.com/operator-framework/operator-registry/blob/master/docs/design/operator-bundle.md#bundle-annotations Because of this `com.redhat.openshift.versions` which is OCP specific is not synched like this ones.

See that sync all annotations to the bundle.Dockerfile is a logic much more intuitive for users ( shows for me a better ux), simple and easy to keep maintained. It is exactly what I would expect checking the bundle.Dockerfile with labels for the default annotations. It shows not intuitive at all to be aware that has a defined collection that is synched only.  

So, can you see and share any reason/argumentation for us not to change it to ensure that any annotation will end up with a Label in the bundle.Dockerfile? 

c/c @fabianvf @jmrodri

File: internal/cmd/operator-sdk/generate/bundle/cmd.go [hunk 1]
```diff
@@ -129,6 +130,8 @@ func (c *bundleCmd) addFlagsTo(fs *pflag.FlagSet) {
 		"This option can only be used if --deploy-dir is set")
 	fs.StringVar(&c.kustomizeDir, "kustomize-dir", filepath.Join("config", "manifests"),
 		"Directory containing kustomize bases in a \"bases\" dir and a kustomization.yaml for operator-framework manifests")
+	fs.StringVar(&c.customAnno, "custom-annotations", filepath.Join("config", "metadata"),
+		"Directory containing custom annotations for operator bundle metadata")
```
Comment (camilamacedo86) [inline internal/cmd/operator-sdk/generate/bundle/cmd.go:134]: I think we should not create a new dir / layout

Comment (camilamacedo86) [reply internal/cmd/operator-sdk/generate/bundle/cmd.go:134]: See: https://github.com/operator-framework/operator-sdk/pull/5560#issuecomment-1049202882

**Comment by andreasgerstmayr** [comment]
Why was this PR closed, is there another solution for adding the `com.redhat.openshift.versions` label and annotation?

**Pull request closed without merging**